File: frontend/src/lib/api/library.ts
/**
 * Document corpus and library endpoints.
 *
 * Two different searches live here and they are not interchangeable:
 *
 * * ``/documents/search`` queries the *federated* scholarly sources (arXiv,
 *   Crossref, Gutenberg, Open Library, Wikipedia, Internet Archive) and returns
 *   results to download into the local corpus;
 * * ``/library/search`` queries the *local* corpus — the PDFs already downloaded
 *   and indexed with SQLite FTS5 — and returns page-level hits with snippets.
 */

import * as api from "./client";
import { LibraryHit, LibraryResponse } from "@/types/library";
import { DocumentResult } from "@/types/research";

/** ``GET /documents/search`` — federated search across the six public sources. */
export function searchDocuments(q: string, limit = 10): Promise<DocumentResult[]> {
  return api.apiFetch<DocumentResult[]>("/documents/search", {
    query: { q, limit },
  });
}

/** ``GET /library/search`` — full-text search over the local corpus. */
export function searchLibrary(q: string, limit = 20): Promise<LibraryHit[]> {
  return api.apiFetch<LibraryHit[]>("/library/search", {
    query: { q, limit },
  });
}

/** ``GET /library`` — the indexed documents plus corpus counters. */
export function getLibrary(): Promise<LibraryResponse> {
  return api.apiFetch<LibraryResponse>("/library");
}

File: src/content_factory/api/routers/library.py
"""Federated document search and the curated local library."""

from __future__ import annotations

from fastapi import APIRouter

from ...models import (
    DocumentResult,
    LibraryHit,
)
from ...service import ContentFactoryService


def build_router(service: ContentFactoryService) -> APIRouter:

    router = APIRouter()

    @router.get("/documents/search", response_model=list[DocumentResult])
    async def search_documents(q: str, limit: int = 10) -> list[DocumentResult]:
        return await service.search_documents(q, limit=max(1, min(limit, 50)))

    @router.get("/library/search", response_model=list[LibraryHit])
    async def search_library(q: str, limit: int = 20) -> list[LibraryHit]:
        return service.search_library(q, limit=max(1, min(limit, 100)))

    @router.get("/library")
    async def library() -> dict:
        return {
            "documents": service.list_library(),
            "stats": service.library_stats(),
        }

    return router
